### PR TITLE
chore: consolidate payload types for user sign-up/log-in activities

### PR DESCRIPTION
**Pull request by tianrunhe**
# Description

I want to consolidate the response payload types for sign-up/login related mutations so it's easier to track sign up for GA4 from client side. This PR:

- Add a new response payload `UserLogInPayload`
- Replace `LoginWithGooglePayload` with `UserLogInPayload`
- Replace `SignUpWithPasswordPayload` with `UserLogInPayload`
- Replace `LoginWithPasswordPayload` with `UserLogInPayload`
- Replace `VerifyEmailPayload` with `UserLogInPayload`

## Final checklist

- [x] I checked the [code review guidelines](../docs/codeReview.md)
- [x] I have added [Metrics Representative](../docs/codeReview.md#metrics-representative) as reviewer(s) if my PR invovles metrics/data/analytics related changes
- [x] I have performed a self-review of my code, the same way I'd do it for any other team member
- [x] I have tested all cases I listed in the testing scenarios and I haven't found any issues or regressions
- [x] Whenever I took a non-obvious choice I added a comment explaining why I did it this way
- [x] I added the label `One Review Required` if the PR introduces only minor changes, does not contain any architectural changes or does not introduce any new patterns and I think one review is sufficient'
- [x] PR title is human readable and could be used in changelog

### CODE REVIEW

**Review by tianrunhe — COMMENTED** [review]
The large portion of diffs come from files moving:

- packages/server/graphql/mutations/loginWithPassword.ts -> packages/server/graphql/public/mutations/loginWithPassword.ts
- packages/server/graphql/mutations/signUpWithPassword.ts -> packages/server/graphql/public/mutations/signUpWithPassword.ts
- packages/server/graphql/mutations/verifyEmail.ts -> packages/server/graphql/public/mutations/verifyEmail.ts

Each mutation is also modified to fit the new mutation pattern and the rate limits are moved to permissions.ts

File: packages/server/graphql/public/mutations/loginWithPassword.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import encodeAuthToken from '../../../utils/encodeAuthToken'
+import {MutationResolvers} from '../resolverTypes'
+import attemptLogin from '../../mutations/helpers/attemptLogin'
+
+const loginWithPassword: MutationResolvers['loginWithPassword'] = async (
+  _source,
+  {email, password},
+  context
+) => {
+  if (process.env.AUTH_INTERNAL_DISABLED === 'true') {
```
Comment (Dschoordsch) [inline packages/server/graphql/public/mutations/loginWithPassword.ts:11]: -1 You did a nice job moving this to `permissions.ts` for `signUpWithPassword`, any reason this should stay here for this mutation?

**Review by Dschoordsch — CHANGES_REQUESTED** [review]
Did not test yet

File: packages/server/graphql/public/mutations/loginWithPassword.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import {AuthenticationError} from 'parabol-client/types/constEnums'
+import encodeAuthToken from '../../../utils/encodeAuthToken'
+import {MutationResolvers} from '../resolverTypes'
+import attemptLogin from '../../mutations/helpers/attemptLogin'
+
+const loginWithPassword: MutationResolvers['loginWithPassword'] = async (
```
[the remark this answers is not in the document]
Comment (tianrunhe) [reply packages/server/graphql/public/mutations/loginWithPassword.ts:6]: Done!

File: packages/server/graphql/public/mutations/loginWithPassword.ts (hunk 1 above)
Comment (tianrunhe) [reply packages/server/graphql/public/mutations/loginWithPassword.ts]: I overlooked this file! Moved over.

**Review by Dschoordsch — APPROVED** [review]
Did not test SSO

**Pull request merged**